File: javascript/easy/strobogrammaticNumber.js
/**
 * @param {string} num
 * @return {boolean}
 */
var isStrobogrammatic = function(num) {
    let map = { '0': '0', '1': '1', '8': '8', '6': '9', '9': '6' };
    let upsideDown =  "";
    for(let c of num) {
        upsideDown = map[c] + upsideDown;
    }
    return upsideDown === num;
};

File: ruby/easy/strobogrammatic_number.rb
# @param {String} num
# @return {Boolean}
def is_strobogrammatic(num)
    map = { :"0" => "0", :"1" => "1", :"8" => "8", :"6" => "9", :"9" => "6" }
    upside_down = ""
    num.each_char do |c|
        upside_down = (map[c.to_sym] ? map[c.to_sym] : "") + upside_down
    end
    return num == upside_down
end
